### PR TITLE
docs/k8s: add some high-level operator architecture diagrams

### PR DESCRIPTION
**Pull request by tomhjp**
This is an experiment to see how useful we will find it to have some text-based diagrams to document how various components of the operator work. There are no plans to link to this from elsewhere yet, but hopefully it will be a useful reference internally.

Updates #cleanup

### CODE REVIEW

File: docs/k8s/operator-architecture.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
+# Operator architecture diagrams
+
+The Tailscale Kubernetes operator has a collection of use-cases that can be
```
Comment (irbekrm) [inline docs/k8s/operator-architecture.md:3]: Suggestion: link to the operator docs

File: docs/k8s/operator-architecture.md [hunk 2]
```diff
@@ -50,9 +66,14 @@ flowchart LR
...
-to the Service, and the operator will create an ingress proxy that allows
-devices anywhere on the tailnet to access the Service.
+that selects the app's Pods. Either add the annotation
+`tailscale.com/expose: "true"` or specify `.spec.type` as `Loadbalancer` and
+`.spec.loadBalancerClass` as `tailscale`. The operator will create an ingress
+proxy that allows devices anywhere on the tailnet to access the Service.
+
+The proxy Pod uses `iptables` or `nftables` rules to DNAT traffic bound for the
+proxy's tailnet IP to the Service's internal Cluster IP instead. This usually
+means traffic coming in on `tailscale0` is forwarded to the `eth0` interface.
```
Comment (irbekrm) [inline docs/k8s/operator-architecture.md:76]: Suggestion:
```suggestion
The proxy Pod uses `iptables` or `nftables` rules to DNAT traffic bound for the
proxy's tailnet IP to the Service's internal Cluster IP instead. 
```
(altough in practice usually true, it doesn't really add anything to the previous sentence and seems a bit confusing)

**Review by irbekrm — APPROVED** [review]
Looks great to me, thank you for working on this.

I've added two suggestions.

A few more nits:
- it'd be nice if the whole of data path was in a different color from the other arrows.
I think especially in egress case, it would be hard for user to understand where the traffic originates (the arrow structure might be easy to miss)
- it'd be nice if ProxyGroup too had an example data path

In general, looks great though, glad this is going in

**Comment by tomhjp** [comment]
> it'd be nice if the whole of data path was in a different color from the other arrows.

I went for red for WireGuard because the logo is red, and blue for "Other network traffic"

> it'd be nice if ProxyGroup too had an example data path

Added, but maybe worth another look as this added some more technical details that I'd appreciate review on if you get a moment please.

**Comment by tomhjp** [comment]
Thanks for the reviews!

**Pull request merged**